core/chains/evm/gas/suggested_price_estimator.go

```diff
@@ -177,8 +188,46 @@ func (o *SuggestedPriceEstimator) GetLegacyGas(ctx context.Context, _ []byte, Ga
 	return
 }
 
-func (o *SuggestedPriceEstimator) BumpLegacyGas(_ context.Context, _ *assets.Wei, _ uint32, _ *assets.Wei, _ []EvmPriorAttempt) (bumpedGasPrice *assets.Wei, chainSpecificGasLimit uint32, err error) {
-	return nil, 0, errors.New("bump gas is not supported for this chain")
+// Refresh the gas price in case the current one has gone stale.
+// The only reason bumping logic would be called on the SuggestedPriceEstimator is if there was a significant price spike
+// between the last price update and when the tx was submitted. Refreshing the price helps ensure the latest market changes are accounted for.
+func (o *SuggestedPriceEstimator) BumpLegacyGas(ctx context.Context, originalFee *assets.Wei, feeLimit uint32, maxGasPriceWei *assets.Wei, _ []EvmPriorAttempt) (newGasPrice *assets.Wei, chainSpecificGasLimit uint32, err error) {
+	chainSpecificGasLimit = feeLimit
+	ok := o.IfStarted(func() {
+		// Immediately return error if original fee is greater than or equal to the max gas price
+		// Prevents a loop of resubmitting the attempt with the max gas price
+		if originalFee.Cmp(maxGasPriceWei) >= 0 {
+			err = fmt.Errorf("original fee (%s) greater than or equal to max gas price (%s) so cannot be bumped further", originalFee.String(), maxGasPriceWei.String())
+			return
+		}
+		err = o.forceRefresh(ctx)
+		if newGasPrice = o.getGasPrice(); newGasPrice == nil {
+			err = errors.New("failed to refresh and return gas; gas price not set")
+			return
+		}
+		o.logger.Debugw("GasPrice", newGasPrice, "GasLimit", feeLimit)
+	})
+	if !ok {
+		return nil, 0, errors.New("estimator is not started")
+	} else if err != nil {
+		return
+	}
+	if newGasPrice != nil && newGasPrice.Cmp(maxGasPriceWei) > 0 {
```

Comment:
No need to return error here, we can let it cap on the max price down below. Just add a log below if the price gets capped so we get alerted.

Reply:
I think we only want to cap at the max gas price if the price from the RPC is still lower than the max. In that scenario, we'd still be using a price that's equal or above the one recommended by the RPC. This clause checks if the price from the RPC is higher than the max gas price. If it is then capping at the max gas price would submit a tx below the recommended price which wouldn't succeed. I was thinking to treat this the same as the other estimators that have bumped past the max and return an error.

Reply:
So here our maxGasPriceWei is lower than what the chain recommends.
Either we hard error, or keep bumping at maxGasPrice.
I think hard error might be better, to get NOP's attention to the bad config, instead of masking that error, and sending a Tx that we know is underpriced.

I see this as a bad configuration by the NOP.